tools/rna_tools/targetfinder/TargetFinder_threads.xml

```diff
@@ -0,0 +1,77 @@
+<tool id="targetfinder_threads" name="Plant small RNA target prediction tool" version="0.1.0">
+    <requirements>
+        <requirement type="package" version="1.07">targetfinder</requirement>
+        <requirement type="package" version="35.4.12">fasta</requirement>
+    </requirements>
+    <stdio>
+        <exit_code range="1:" />
+    </stdio>
+
+    <command><![CDATA[
+        targetfinder_threads.pl
+            -f $f
+            -d $d
+            -c $c
+            -t "\${GALAXY_SLOTS:-12}"
+            $r
+            -o $output1
+    ]]></command>
+    <inputs>
+        <param label="Input small RNA sequences file" help="(-f) FASTA-format" name="f" type="data" format="fasta" />
+        <param label="Target sequence database file" help="(-d) FASTA-format" name="d" type="data" format="fasta" />
+        <param label="Prediction score cutoff value" help="(-c) DEFAULT = 4" name="c" type="float" value="4.0" />
+        <param label="Search reverse strand for targets?" help="(-r) Use this option if the database is genomic DNA." name="r" type="boolean" falsevalue="" truevalue="-r" checked="false" />
+    </inputs>
+    <outputs>
+        <data name="output1" format="data"/>
+    </outputs>
+    <tests>
+        <test>
+            <!-- Order of predicted targets are not constant in output per run with identical parameter. -->
+            <!-- This makes test fail while the wrapper actually works (external validation). -->
+            <param name="f" value="ath_miRNAs_test.fa"/>
```

Comment:
A book, maybe not a bad idea :)
For the time being please look at this project:
https://planemo.readthedocs.org/en/latest/writing.html
 and this wiki page
https://wiki.galaxyproject.org/Admin/Tools/ToolConfigSyntax